File: backend/src/main/java/com/bytechat/services/DMRequestService.java
package com.bytechat.services;

import com.bytechat.entity.DMRequest;
import com.bytechat.entity.User;

import java.util.List;

public interface DMRequestService {
    DMRequest sendRequest(Long workspaceId, User sender, Long receiverId);
    DMRequest acceptRequest(User receiver, Long requestId);
    void rejectRequest(User receiver, Long requestId);
    List<DMRequest> getPendingRequests(User user);
}

File: frontend/src/services/dmRequestService.js
import api from './api';

const dmRequestService = {
  async sendRequest(receiverId) {
    const response = await api.post(`/dm/requests/send/${receiverId}`);
    return response.data;
  },
  async acceptRequest(requestId) {
    const response = await api.post(`/dm/requests/accept/${requestId}`);
    return response.data;
  },
  async rejectRequest(requestId) {
    const response = await api.post(`/dm/requests/reject/${requestId}`);
    return response.data;
  },
  async getPendingRequests() {
    const response = await api.get('/dm/requests/pending');
    return response.data;
  }
};

export default dmRequestService;
